eureka-client/src/test/java/com/netflix/discovery/util/EurekaEntityFunctionsTests.java

```diff
@@ -0,0 +1,363 @@
+        application.addInstance(instanceInfo);
+
+        Assert.assertNotEquals(application.getInstances(),
+                EurekaEntityFunctions.mergeApplication(
+                        application, application).getInstances());
+    }
+
+    @Test
+    public void testUpdateMetaIfNotNullReturnApplications() {
+        InstanceInfo instanceInfo = Mockito.mock(InstanceInfo.class);
+        Mockito.when(instanceInfo.getId()).thenReturn("foo");
+        Mockito.when(instanceInfo.getStatus())
+                .thenReturn(InstanceInfo.InstanceStatus.UP);
+        Application application = new Application("foo");
+        application.addInstance(instanceInfo);
+
+        ArrayList<Application> applicationArrayList = new ArrayList<>();
+        applicationArrayList.add(application);
+        Applications applications = new Applications(
+                "appsHashCode", 1559658285l, applicationArrayList);
+
+        Assert.assertEquals(applications.getRegisteredApplications(),
+                EurekaEntityFunctions.updateMeta(
+                        applications).getRegisteredApplications());
+    }
+
+    @Test
+    public void testCountInstancesIfApplicationsHasInstancesReturnSize() {
+        InstanceInfo instanceInfo = Mockito.mock(InstanceInfo.class);
+        Mockito.when(instanceInfo.getId()).thenReturn("foo");
+        Application application = new Application("foo");
+        application.addInstance(instanceInfo);
+
+        ArrayList<Application> applicationArrayList = new ArrayList<>();
+        applicationArrayList.add(application);
+        Applications applications = new Applications(
+                "appsHashCode", 1559658285l, applicationArrayList);
+        applications.addApplication(application);
+
+        Assert.assertEquals(2,
+                EurekaEntityFunctions.countInstances(applications));
+    }
+
+    @Test
+    public void testComparatorByAppNameAndIdIfNotNullReturnInt() {
+        InstanceInfo instanceInfo1 = Mockito.mock(InstanceInfo.class);
+        InstanceInfo instanceInfo2 = Mockito.mock(InstanceInfo.class);
+        InstanceInfo instanceInfo3 = Mockito.mock(InstanceInfo.class);
+        InstanceInfo instanceInfo4 = Mockito.mock(InstanceInfo.class);
+        Mockito.when(instanceInfo3.getAppName()).thenReturn("foo");
+        Mockito.when(instanceInfo3.getId()).thenReturn("foo");
+        Mockito.when(instanceInfo4.getAppName()).thenReturn("bar");
+        Mockito.when(instanceInfo4.getId()).thenReturn("bar");
+
+        Assert.assertEquals(1, EurekaEntityFunctions.comparatorByAppNameAndId()
+                .compare(instanceInfo1, instanceInfo2));
+        Assert.assertEquals(1, EurekaEntityFunctions.comparatorByAppNameAndId()
+                .compare(instanceInfo3, instanceInfo2));
+        Assert.assertEquals(-1, EurekaEntityFunctions.comparatorByAppNameAndId()
+                .compare(instanceInfo1, instanceInfo3));
+        Assert.assertEquals(4, EurekaEntityFunctions.comparatorByAppNameAndId()
```

Comment:
Change these 1, -1, 4 and so on to greater than / less than and 0 since it's what the underlying API contract is, i.e. it does not return 1, it returns a positive integer

Reply:
I am not sure if I did this as expected

Reply:
Yes it's correct now, thanks.